packages/api/src/platforms/vtex/utils/shouldUpdateShippingData.ts

```diff
@@ -0,0 +1,50 @@
+import { IStoreSession } from '../../../__generated__/schema'
+import { OrderForm } from '../clients/commerce/types/OrderForm'
+
+export const shouldUpdateShippingData = (
+  orderForm: OrderForm,
+  session: IStoreSession
+) => {
+  if (
+    orderForm.items.length === 0 ||
+    !orderForm.shippingData ||
+    !orderForm.shippingData.logisticsInfo ||
+    orderForm.shippingData.logisticsInfo.length === 0
+  ) {
+    return false
+  }
+
+  const { address, logisticsInfo } = orderForm.shippingData
+
+  const hasDifferentPostalCode =
```

Comment:
Return right after to avoid having to run all the code below in case this is true